08_education/preschool_county.qmd

```diff
@@ -0,0 +1,766 @@
+```
+
+## Crosswalk
+
+Read in the PUMA to county crosswalk file. This file is created by the program generate_puma_county_crosswalks.rmd in the geographic-crosswalks folder. 
+```{r}
+puma_county_crosswalk <- read_csv(here::here("geographic-crosswalks", "data", "crosswalk_puma_to_county.csv"))
+```
+
+Crate a version of the crosswalk with just the county-level values of data quality variables for each crosswalk period.
+```{r}
+county_puma <- puma_county_crosswalk %>%
+  group_by(crosswalk_period, statefip, county) %>% 
+  summarise(geographic_allocation_quality = mean(geographic_allocation_quality)) %>% 
+  ungroup()
+```
+
+## Merge Crosswalk 
+
+Create a variable in the ACS data indicating whether the survey is pre or post-2022. This will be used to join on the correct crosswalk information.
+
+```{r}
+acs_clean <- acs_clean %>% 
+  mutate(crosswalk_period = ifelse(year < 2022, "pre-2022", "2022"))
+
+```
+
+Check that the assignment worked. All years prior to 2022 should be designated "pre-2022".
+
+```{r}
+count(acs_clean, crosswalk_period, year)
+
+```
+
+Join the cleaned ACS data onto the crosswalk.
+
+```{r}
+acs_crosswalked <- left_join(
+  acs_clean, 
+  puma_county_crosswalk, 
+  by = c("crosswalk_period", "statefip","puma"),
+  relationship = "many-to-many"
+)
+
+rm(acs_clean)
+
+```
+
+Drop any observations with NA for `afact` (i.e. there is no county of interest overlapping this PUMA).
+
+```{r}
+count(acs_crosswalked, is.na(afact))
+
+acs_crosswalked <- acs_crosswalked %>% 
+  drop_na(afact)
+
+```
+
+Also filter out cases where `afact` is equal to zero. These cases will not be counted in the metric calculation.
+
+```{r}
+count(acs_crosswalked, afact == 0)
+
+acs_crosswalked <- acs_crosswalked %>% 
+  filter(afact > 0)
+
+```
+
+Adjust the person weight to account for PUMA-to-county mapping (those where the PUMA is not entirely inside the county).
+
+```{r}
+select(acs_crosswalked, perwt, repwtp1, repwtp80, afact)
+
+acs_crosswalked <- acs_crosswalked %>%
+  mutate(perwt = perwt * afact) %>% 
+  mutate(across(matches("repwtp[0-9]+"), ~.x * afact))
+
+select(acs_crosswalked, perwt, repwtp1, repwtp80, afact) 
+```
+
+Create a binary variable to distinguish between 1 and 5 year ACS surveys. 5-year samples include the term "5-year".
+
+```{r}
+acs_crosswalked <- acs_crosswalked %>%
+  mutate(sample_structure = ifelse(str_detect(sample, "5-year"), "5year", "1year")) 
+
+```
+
+### Create computational variables
+
+Create a binary preschool variable indicating whether the child is in preschool/pre-K or not. IPUMS codes  Nursery school/preschool (1) and N/A (0) (re: grade level attending). 
+The way the code was built in prior years it is assumed N/A means that the child is not attending any school.
+```{r}
+acs_crosswalked <- acs_crosswalked %>% 
+  filter(grade_attending %in% c("Nursery school/preschool", "N/A")) %>% 
+  mutate(preschool = ifelse(grade_attending == "Nursery school/preschool", 1, 0)) 
```

Comment:
Note, you don't actually need `ifelse()` in these cases. 

```r
mutate(preschool = grade_attending == "Nursery school/preschool")
```

will create `TRUE`/`FALSE`, which are treated as `1`/`0`.

Reply:
This is awesome, did not know!